oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+
+.. note:: Although this example includes versions, based on the decisions below our changelog entries would typically only include dates, because our versioned changelogs are managed using GitHub Releases.
+
+::
+
+  Changelog
+  ===========
+
+  All notable changes to this project will be documented in this file.
+
+  The format is based on recommendations in `OEP-44`_.
+  This project adheres to `Semantic Versioning`_.
+
+  .. _OEP-44: https://open-edx-proposals.readthedocs.io/en/latest/oep-0044-bp-changelogs.html
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+  [2.0.0] - 2020-03-21
+  --------------------
+
+  Removed
+  ~~~~~~~
+
+  * Removed deprecated function `xxx`. See DEPR-XXX for details.
+
+  [2.0.0] - 2020-03-21
+  --------------------
+
+  Added
+  ~~~~~
+
+  * Feature x. See announcement here: https://xxx
+
+Notable Differences
+~~~~~~~~~~~~~~~~~~~
+
+The following are notable differences between the recommendations of this OEP and the `keepachangelog.com`_ recommendations:
+
+* We use GitHub Releases in certain cases, as you'll see below, for developer efficiency purposes.
+* Because we typically release every change for libraries, we don't get any benefit from an "Unreleased" section.
+* We sometimes use automation that uses git commit messages for changelog messages. As noted above, it is acceptable and appropriate to clean these up as needed.
+
+Library Repositories
+--------------------
+
+All Open edX library repositories should contain a CHANGELOG.rst. These libraries should all adhere to `Semantic Versioning`_.
+
+Semantic Versioning
+~~~~~~~~~~~~~~~~~~~
+
+All versioned code should use a MAJOR.MINOR.PATCH format for versioning, as defined by the `Semantic Versioning v2`_ specification. Here is a copy of the most relevant part of the summary:
+
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+* *MAJOR* version when you make *incompatible API changes*,
+* *MINOR* version when you *add functionality in a backwards compatible manner*, and
+* *PATCH* version when you make *backwards compatible bug fixes*.
+
+Every published version should have a corresponding entry in the changelog, but not necessarily in CHANGELOG.rst.
+
+One of the most important rules when choosing whether to increment the MAJOR, MINOR, or PATCH version, is that backward incompatible changes always result in a new MAJOR version, and never a MINOR or PATCH version.
+
+* You must increment the MAJOR version for backward incompatible changes, even if you think it is unlikely to affect anyone. You can simply note in the changelog entry why it is backward incompatible and why you think it is unlikely to affect anyone.
+* If a backward incompatible change is found in a MINOR or PATCH version after publishing, first update the changelog and communicate immediately.  Next, a new MINOR or PATCH version should immediately be released which reverts this change, and it should be moved to a new MAJOR version.
```

Comment:
I wonder if versioning logic like this belongs in this OEP.  It could become a more difficult part to get agreement on.

Reply:
By "versioning logic", do you mean these process steps?  I updated the language to make it less about what you must do, and more as help for what you might do.

Or, do you mean the entire topic of semantic versioning?  I don't expect a lot of disagreement there, but I am happy to move this to a separate OEP if you think that makes more sense.

Reply:
We can wait to see what further review brings. The mechanics of semantic versioning feels orthogonal to how changelogs should be managed, but I can see that the topics are related.